libs/ledger/include/ledger/chain/block.hpp

```diff
@@ -113,27 +114,29 @@ struct MapSerializer<ledger::Block, D>
   static uint8_t const MINER_SIGNATURE = 5;
   static uint8_t const HASH            = 6;
   static uint8_t const PREVIOUS_HASH   = 7;
-  static uint8_t const MERKLE_HASH     = 8;
-  static uint8_t const BLOCK_NUMBER    = 9;
-  static uint8_t const MINER           = 10;
-  static uint8_t const MINER_ID        = 11;
-  static uint8_t const LOG2_NUM_LANES  = 12;
-  static uint8_t const SLICES          = 13;
-  static uint8_t const DAG_EPOCH       = 14;
-  static uint8_t const TIMESTAMP       = 15;
-  static uint8_t const ENTROPY         = 16;
+  static uint8_t const NEXT_HASH       = 8;
+  static uint8_t const MERKLE_HASH     = 9;
+  static uint8_t const BLOCK_NUMBER    = 10;
+  static uint8_t const MINER           = 11;
+  static uint8_t const MINER_ID        = 12;
+  static uint8_t const LOG2_NUM_LANES  = 13;
+  static uint8_t const SLICES          = 14;
+  static uint8_t const DAG_EPOCH       = 15;
+  static uint8_t const TIMESTAMP       = 16;
+  static uint8_t const ENTROPY         = 17;
 
   template <typename Constructor>
   static void Serialize(Constructor &map_constructor, Type const &block)
   {
-    auto map = map_constructor(16);
+    auto map = map_constructor(17);
     map.Append(NONCE, block.nonce);
     map.Append(PROOF, block.proof);
     map.Append(WEIGHT, block.weight);
     map.Append(TOTAL_WEIGHT, block.total_weight);
     map.Append(MINER_SIGNATURE, block.miner_signature);
     map.Append(HASH, block.hash);
     map.Append(PREVIOUS_HASH, block.previous_hash);
+    map.Append(NEXT_HASH, block.next_hash);
```

Comment:
To avoid bugs/attacks, this should be cleared when recieveing it in main_chain_rpc_service (the only place blocks can arrive from peers I think)

Reply:
On second thought, it might make more sense to put this clearing in the AddBlock of main chain

Reply:
TLDR: I'm not sure about either.

Previous-to-next relation should be easily verifiable: `∀hash . GetBlock(GetBlock(hash)->next_hash)->previous_hash == hash`; consequently, setting `next_hash` should not be any more malicious than setting `prev` one (which we do trust, I assume). One point at which clearing `next_hash` temporarily could, in theory, make sense is dropping it from the heaviest block of last successfully retrieved chunk if we were unable to retrieve the continuation, the subsequent chain blocks from any source at all, which would probably signify that the whole last chunk was something forged by a malicious peer (who opted later to not send any continuation for its secret malicious purpose) but I'm not sure yet how should this situation be handled at all.

Reply:
It seems to my mind bugprone that you would allow peers to populate this field. As you point out, the last block of a chunk could be invalid, so the main chain needs to track potentially suspicious hashes that are stored within it for the next pull of blocks. But, all next block hashes could also be invalid in that they could have a competing next block, which should cause this field to be empty (?).